apps/prairielearn/src/pages/instructorAssessmentOverrides/instructorAssessmentOverrides.sql

```diff
@@ -0,0 +1,66 @@
+-- BLOCK select_assessment_access_policy
+SELECT
+  COALESCE(format_date_full_compact(aap.created_at, 'America/Chicago'), '—') AS created_at,
+  COALESCE(aap.created_by::text, '—') AS created_by,
+  COALESCE(aap.credit::text, '—') AS credit,
+  COALESCE(format_date_full_compact(aap.end_date, 'America/Chicago'), '—') AS end_date,
+  COALESCE(aap.note, '—') AS note,
+  COALESCE(format_date_full_compact(aap.start_date, 'America/Chicago'), '—') AS start_date,
+  COALESCE((SELECT name from groups where id = aap.group_id),'-') as group_name,
+  COALESCE((SELECT uid FROM users WHERE user_id = aap.user_id),'-') AS student_uid
+FROM
+  assessment_access_policies AS aap
+WHERE assessment_id = $assessment_id
+ORDER BY
+  aap.created_at;
+
+-- BLOCK check_group_in_course_instance
+SELECT
+  COUNT(*) AS count
+FROM
+  groups
+WHERE
+  course_instance_id = $course_instance_id AND id = (SELECT id from groups where name = $group_name)
+
+-- BLOCK insert_assessment_access_policy
+INSERT INTO assessment_access_policies
+  (assessment_id, created_at, created_by, credit, end_date, group_id, note, start_date,user_id)
+VALUES
+  ($assessment_id, $created_at, $created_by, $credit, $end_date, (SELECT id from groups where name = $group_name), $note, $start_date, (SELECT user_id FROM users WHERE uid = $student_uid));
```

Comment:
The select you're using to get the group ID is going to pose a big problem: it will get the IDs of *all* groups across all of PrairieLearn that have this same name.

Rather than doing it inline here, I'd recommend updating `check_group_in_course_instance` to be something more like `select_group_in_assessment` and have it actually return the `groups` row (if it exists). You can still check if the group exists with JavaScript like you already do; you'll then just use the group ID directly in this `insert_assessment_access_policy` query. Let me know if that makes sense!